### ISSUE TITLE
phash of img_hash test fail in java

### ISSUE THREAD

**Issue by lcstore** [opencv/opencv_contrib#1517]
Hi dear:
   I turn the test example of img_hash into java, but test fail.  

   1) env: 
    opencv3.4.0 with moudle img_hash
    JDK8
   2) java test code:
   @Test
    public void testHashCode() {
        List<Integer> codeList = Lists.newArrayList(
                1,0,1,1,1,1,1,1,
                0,1,1,1,1,1,1,1,
                1,1,1,1,1,1,1,1,
                0,1,1,1,1,1,1,1,
                1,1,1,1,1,1,1,1,
                0,1,1,1,1,1,1,1,
                1,1,1,1,1,1,1,1,
                0,1,1,1,1,1,1,1);
        String expect = Joiner.on("").join(codeList);
        int capacity = 32 * 32;
        int value = 0;
        double[] dataArr = new double[capacity];
        for (int i = 0; i != capacity; ++i) {
            dataArr[i] = value++;
        }
        Mat inMat = new Mat(32, 32, CvType.CV_8U);
        inMat.put(0, 0, dataArr);
        Mat outMat = new Mat();
        Img_hash.pHash(inMat, outMat);
        StringBuilder sb = new StringBuilder();
        for (int row = outMat.rows() - 1; row >= 0; row--) {
            for (int col = outMat.cols() - 1; col >= 0; col--) {
                Double val = outMat.get(row, col)[0];
                String origin = "00000000";
                origin = origin + Long.toBinaryString(val.longValue());
                sb.append(origin.substring(origin.length() - 8));
            }
        }
        String binary = sb.toString();
        Assert.assertEquals(expect, binary);
    }

org.junit.ComparisonFailure: 
Expected :1011111101111111111111110111111111111111011111111111111101111111
Actual   :   1111111111111110111111101111111011111110111111101111110011111111
 <Click to see difference>

**Comment by lcstore** [comment]
@stereomatchingkiss can you help me please ?

**Comment by stereomatchingkiss** [comment]
@lcstore  Yes, please wait me a while, I will try to figure out why on this weekend

**Comment by berak** [comment]
also see: http://answers.opencv.org/question/182068/phash-of-img_hash-test-fail-in-java/

**Comment by stereomatchingkiss** [comment]
@lcstore  Do you solve your problem? I think berak give you the answer you need.
